mm/mm_heap/mm.h

```diff
@@ -130,10 +130,10 @@
 
 #ifdef CONFIG_MM_SMALL
 # define MM_ALLOC_BIT    0x8000
-# define MMSIZE_MAX      UINT16_MAX
+# define MMSIZE_MAX      (UINT16_MAX >> 1)
```

Comment:
it's better to move the special bit to the low end since heap always allocate the memory at least 2bytes alignment.

Reply:
Done, and i have tested (mm test) this patch with CONFIG_MM_SAMLL enable/disable, do you think more tests are needed?